src/dmd/doc.d

```diff
@@ -2142,9 +2158,13 @@ private bool isReservedName(const(char)[] str)
 /**************************************************
  * Highlight text section.
  */
-private void highlightText(Scope* sc, Dsymbols* a, OutBuffer* buf, size_t offset)
+private void highlightText(Scope* sc, Dsymbols* a, Loc loc, OutBuffer* buf, size_t offset)
```

Comment:
Please add `Params:` section. Adding parameters is a good time to do it.

Reply:
Curiously, `loc` is mutable (it is normally `const`). This needs explanation.

Reply:
The mutable loc is to allow incrementing its `linenum`. The reason for this is the same reason `escapeStrayParenthesis` already does it: to allow printing the correct line number when an error is encountered in a multiline block of ddoc.

Reply:
By "needs explanation" I mean it needs it in the `Params:` section.

Reply:
I'm not sure I understand; it's probably my ignorance of the rest of the dmd codebase. A Loc struct is passed into several functions in doc.d and in no case is it `const`. Since it's passed by value any changes `highlightText` makes to `linenum` won't escape the function scope. I'm not sure what I should document here.

Reply:
A lot of older code in dmd passes `loc` by mutable value, a lot of that has been improved to passing it by `const ref`. Almost never is it needed to be mutable, so such is surprising and should be documented, such as:
```
Params:
    loc = source location of start of text. It is a mutable copy to allow incrementing its linenum, for printing the correct line number when an error is encountered in a multiline block of ddoc.
```

Reply:
`const ref` makes much more sense. I'll go ahead and steal your doc :)